pyquil/quilbase.py

```diff
@@ -131,30 +134,51 @@ def _convert_to_rs_instruction(instr: AbstractInstruction) -> quil_rs.Instructio
         return quil_rs.Instruction.from_delay(instr)
     if isinstance(instr, quil_rs.Fence):
         return quil_rs.Instruction.from_fence(instr)
+    if isinstance(instr, quil_rs.FrameDefinition):
+        return quil_rs.Instruction.from_frame_definition(instr)
     if isinstance(instr, quil_rs.Gate):
         return quil_rs.Instruction.from_gate(instr)
+    if isinstance(instr, quil_rs.GateDefinition):
+        return quil_rs.Instruction.from_gate_definition(instr)
     if isinstance(instr, quil_rs.MeasureCalibrationDefinition):
         return quil_rs.Instruction.from_measure_calibration_definition(instr)
     if isinstance(instr, quil_rs.Measurement):
         return quil_rs.Instruction.from_measurement(instr)
     if isinstance(instr, quil_rs.Pragma):
         return quil_rs.Instruction.from_pragma(instr)
+    if isinstance(instr, quil_rs.Pulse):
+        return quil_rs.Instruction.from_pulse(instr)
+    if isinstance(instr, quil_rs.RawCapture):
+        return quil_rs.Instruction.from_raw_capture(instr)
     if isinstance(instr, quil_rs.Reset):
         return quil_rs.Instruction.from_reset(instr)
-    if isinstance(instr, AbstractInstruction):
-        return quil_rs.Instruction(instr)
-    else:
-        raise ValueError(f"{type(instr)} is not an Instruction")
+    if isinstance(instr, quil_rs.SetFrequency):
+        return quil_rs.Instruction.from_set_frequency(instr)
+    if isinstance(instr, quil_rs.SetPhase):
+        return quil_rs.Instruction.from_set_phase(instr)
+    if isinstance(instr, quil_rs.SetScale):
+        return quil_rs.Instruction.from_set_scale(instr)
+    if isinstance(instr, quil_rs.ShiftFrequency):
+        return quil_rs.Instruction.from_shift_frequency(instr)
+    if isinstance(instr, quil_rs.ShiftPhase):
+        return quil_rs.Instruction.from_shift_phase(instr)
+    if isinstance(instr, quil_rs.SwapPhases):
+        return quil_rs.Instruction.from_swap_phases(instr)
+    if isinstance(instr, quil_rs.WaveformDefinition):
+        return quil_rs.Instruction.from_waveform_definition(instr)
+    raise ValueError(f"{type(instr)} is not an Instruction")
 
 
-def _convert_to_rs_instructions(instrs: Iterable[AbstractInstruction]) -> List[quil_rs.Instruction]:
+def _convert_to_rs_instructions(instrs: Sequence[AbstractInstruction]) -> List[quil_rs.Instruction]:
```

Comment:
I switched any function parameter with the type of `List` or `Iterable` to `Sequence`. Both `list` and any `Iterable` is compatible with `Sequence`. `Iterable` at times needlessly restrictive, and the `List` type is invariant, meaning that `List[T]` could only contain exactly `T`, no sub or super types of `T`.